cocos/renderer/gfx-metal/MTLCommandBuffer.mm

```diff
@@ -507,24 +523,67 @@ of this software and associated engine source code (the "Software"), a limited,
         }
 
         if (sampler.stages & ShaderStageFlagBit::VERTEX) {
-            _commandEncoder.setVertexTexture(gpuDescriptor.texture->getMTLTexture(), sampler.textureBinding);
-            _commandEncoder.setVertexSampler(gpuDescriptor.sampler->getMTLSamplerState(), sampler.samplerBinding);
+            _renderEncoder.setVertexTexture(gpuDescriptor.texture->getMTLTexture(), sampler.textureBinding);
+            _renderEncoder.setVertexSampler(gpuDescriptor.sampler->getMTLSamplerState(), sampler.samplerBinding);
         }
 
         if (sampler.stages & ShaderStageFlagBit::FRAGMENT) {
-            _commandEncoder.setFragmentTexture(gpuDescriptor.texture->getMTLTexture(), sampler.textureBinding);
-            _commandEncoder.setFragmentSampler(gpuDescriptor.sampler->getMTLSamplerState(), sampler.samplerBinding);
+            _renderEncoder.setFragmentTexture(gpuDescriptor.texture->getMTLTexture(), sampler.textureBinding);
+            _renderEncoder.setFragmentSampler(gpuDescriptor.sampler->getMTLSamplerState(), sampler.samplerBinding);
+        }
+        
+        if(sampler.stages & ShaderStageFlagBit::COMPUTE) {
+            _computeEncoder.setTexture(gpuDescriptor.texture->getMTLTexture(), sampler.textureBinding);
         }
     }
 }
 
 void CCMTLCommandBuffer::blitTexture(Texture *srcTexture, Texture *dstTexture, const TextureBlit *regions, uint count, Filter filter) {
+    id<MTLBlitCommandEncoder> encoder = [_mtlCommandBuffer blitCommandEncoder];
+    id<MTLTexture> src = ((CCMTLTexture*)srcTexture)->getMTLTexture();
+    id<MTLTexture> dst = nil;
+    CCMTLDevice* device = static_cast<CCMTLDevice*>(_device);
+    if(!dstTexture) {
+        CAMetalLayer* currentDrawable = static_cast<CAMetalLayer*>(device->getCurrentDrawable());
+        dst = [currentDrawable texture];
+    } else {
+        dst = ((CCMTLTexture*)dstTexture)->getMTLTexture();
+    }
```

Comment:
Can modify current drawable? When need to do like this?

Reply:
gfxtestcase:
```
    if (_device->hasFeature(gfx::Feature::COMPUTE_SHADER)) {
        commandBuffer->pipelineBarrier(_globalBarriers[1], _textureBarriers.data(), _textures.data(), 1);

        commandBuffer->bindPipelineState(_compPipelineState);
        commandBuffer->bindDescriptorSet(0, _compDescriptorSet);
        commandBuffer->dispatch(dispatchInfo);

        commandBuffer->bindPipelineState(_compBGPipelineState);
        commandBuffer->bindDescriptorSet(0, _compBGDescriptorSet);
        commandBuffer->dispatch(bgDispatchInfo);

#if USE_BLIT_TEXTURE
        commandBuffer->pipelineBarrier(_globalBarriers[2], &_textureBarriers[1], _textures.data(), 2);
        commandBuffer->blitTexture(_textures[0], nullptr, &blit, 1u, gfx::Filter::POINT);
#endif
    }
```

when second parameters of blitTexture == null, command buffer blit the texture to swapchain/drawable/framebuffer